pkg/controller/plan/kubevirt.go

```diff
@@ -1166,6 +1175,91 @@ func (r *KubeVirt) virtualMachine(vm *plan.VMStatus) (object *cnv.VirtualMachine
 	return
 }
 
+// Attempt to find a suitable preference.
+func (r *KubeVirt) vmPreference(vm *plan.VMStatus) (virtualMachine *cnv.VirtualMachine, err error) {
+	config, err := r.getOsMapConfig(r.Source.Provider.Type())
+	if err != nil {
+		return
+	}
+	preferenceName, err := r.Builder.PreferenceName(vm.Ref, config)
+	if err != nil {
+		return
+	}
+
+	preferenceName, kind, err := r.getPreference(vm, preferenceName)
+	if err != nil {
+		return
+	}
+
+	virtualMachine = r.emptyVm(vm)
+	virtualMachine.Spec.Preference = &cnv.PreferenceMatcher{Name: preferenceName, Kind: kind}
+	return
+}
+
+func (r *KubeVirt) getOsMapConfig(providerType v1beta1.ProviderType) (configMap *core.ConfigMap, err error) {
+	list := &core.ConfigMapList{}
```

Comment:
listing the config maps here doesn't make much sense - it would be better to propagate the name from the operator (e.g., `vsphere_osmap_configmap_name`) and call 'get'

Reply:
changed to const name of the configmap